File: mobile/lib/models/appointment.dart
class Appointment {
  final int id;
  final int adminId;
  final int? clientId;
  final String date;
  final String startTime;
  final String endTime;
  final String status; // "available", "booked", "cancelled"
  final String? title;
  final String? notes;
  final String? clientName;
  final String? clientEmail;
  final String? companyName;

  Appointment({
    required this.id,
    required this.adminId,
    this.clientId,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.status,
    this.title,
    this.notes,
    this.clientName,
    this.clientEmail,
    this.companyName,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] ?? 0,
      adminId: json['admin_id'] ?? 0,
      clientId: json['client_id'],
      date: json['date']?.toString() ?? '',
      startTime: json['start_time']?.toString() ?? '',
      endTime: json['end_time']?.toString() ?? '',
      status: json['status']?.toString() ?? 'available',
      title: json['title']?.toString(),
      notes: json['notes']?.toString(),
      clientName: json['client_name']?.toString(),
      clientEmail: json['client_email']?.toString(),
      companyName: json['company_name']?.toString(),
    );
  }
}

File: backend/app/models/appointment.py
from __future__ import annotations
from datetime import date, datetime, timezone
from sqlalchemy import String, Text, Date, ForeignKey, Boolean, JSON
from sqlalchemy.orm import Mapped, mapped_column, relationship


from app.core.database import Base


class Appointment(Base):
    __tablename__ = "appointments"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    admin_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    client_id: Mapped[int | None] = mapped_column(ForeignKey("users.id", ondelete="SET NULL"), nullable=True)
    date: Mapped[date] = mapped_column(Date, nullable=False, index=True)
    start_time: Mapped[str] = mapped_column(String(5), nullable=False)  # HH:MM
    end_time: Mapped[str] = mapped_column(String(5), nullable=False)    # HH:MM
    status: Mapped[str] = mapped_column(String(20), default="available", nullable=False)  # available, booked, cancelled, meeting
    title: Mapped[str | None] = mapped_column(String(250))
    notes: Mapped[str | None] = mapped_column(Text)
    meeting_link: Mapped[str | None] = mapped_column(String(500), nullable=True)
    is_group: Mapped[bool | None] = mapped_column(Boolean, default=False, nullable=True)
    attendee_ids: Mapped[list[int] | None] = mapped_column(JSON, nullable=True)
    created_at: Mapped[datetime] = mapped_column(default=lambda: datetime.now(timezone.utc))
    updated_at: Mapped[datetime] = mapped_column(
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc)
    )

    # Relaciones
    admin: Mapped["User"] = relationship("User", foreign_keys=[admin_id], back_populates="appointments_as_admin")
    client: Mapped["User | None"] = relationship("User", foreign_keys=[client_id], back_populates="appointments_as_client")

